### PR TITLE
Development Logger

### LINKED ISSUE #22: Development Logger
Add a logger using [Winston](https://github.com/winstonjs/winston). We're gonna use this in-place of print statements everywhere. It'll give more info + be better formatted.

Since this is development only, we can be log more stuff, so set it's logging level to `debug`. Format is up to you, just pick something reasonable. Have it output to console only.

Put it in Logger directory (See [project structure](https://github.com/uoftblueprint/tces/wiki/Project-Structure))

### PR DESCRIPTION
**Pull request by jordanjanakievski**
Fixes #22 

Creation of a development logger.
We can add the logger to files and output more informational messages instead of `console.log` and label them based on the level of importance.

```
levels = {
  error: 0,
  warn: 1,
  info: 2,
  http: 3,
  verbose: 4,
  debug: 5
};
```

Additional dependencies added
In `server/logger`: winston

### CODE REVIEW

File: server/logger/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		warn: 1,
+		info: 2,
+		http: 3,
+		verbose: 4,
+		debug: 5,
+	};
+
+	for (let level of Object.keys(levels)) {
+		levels[level] = createLogger({
+			levels: { [level]: 0 },
```
Comment (helen-li) [inline server/logger/index.ts:20]: quick question out of curiosity - why do we always put 0 as the value here?

File: server/logger/index.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		error: 0,
+		warn: 1,
+		info: 2,
+		http: 3,
+		verbose: 4,
+		debug: 5,
+	};
+
+	for (let level of Object.keys(levels)) {
+		levels[level] = createLogger({
```
Comment (helen-li) [inline server/logger/index.ts:19]: what are the advantages of creating a logger for each level individually versus creating one logger & changing how we use the logger in our files

**Review by helen-li — COMMENTED** [review]
Looks good overall! The README stuff was really helpful too. Dropped two quick questions :)

File: server/logger/index.ts (hunk 1 above)
Comment (jordanjanakievski) [reply server/logger/index.ts]: I messed around with the code a bit and ended up removing this line.  The formatting function handles the level so this line was no longer needed

File: server/logger/index.ts (hunk 2 above)
Comment (jordanjanakievski) [reply server/logger/index.ts:19]: This is my first time setting up a logger, but I noticed the readability on the server code end is clear as the function will simply state the error type. But if there is a better, more flexible way I would be more than happy to look into and implement it

Comment (helen-li) [reply server/logger/index.ts:19]: gotchu i see! - i was just curious since i don't know much about loggers lol

**Review by helen-li — APPROVED** [review]
thanks jordan lgtm!

**Review by ganeshasapu — APPROVED** [review]
I haven't used loggers before so this was cool to see. Everything works on my end and the ReadMe was very helpful. W PR

File: server/logger/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (EclipseIsDead) [inline server/logger/package.json:1]: I think we only want 1 package.json for the backend, move this to the server package.json

**Review by EclipseIsDead — CHANGES_REQUESTED** [review]
lgtm, small change

File: server/logger/package.json (hunk 3 above)
Comment (jordanjanakievski) [reply server/logger/package.json:1]: I moved the winston dependency to the server package and left the logger free of dependencies

**Review by EclipseIsDead — APPROVED** [review]

**Pull request merged**